### PR TITLE
Editing a model.

### PR DESCRIPTION
**Pull request by gearedmonkey**
### Summary

You can edit a model by clicking the page/pen icon on the view all models screen.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Start server on edit-model-tff111 OR edit-model-story if merged already
2. Start client on this branch
3. View all models and click the pen(cil) to edit the model
4. Witness the form and input any information you feel like editing
5. Verify toast
6. View all models and checkout your changes(does not change the address)

### Links

- [TFF-111](https://msoese.atlassian.net/browse/TFF-111)
- TheFourFifths/consus/pull/41

### CODE REVIEW

File: src/lib/api-client.js [hunk 1]
```diff
@@ -53,6 +53,11 @@ function post(endpoint, data) {
     return call(endpoint, 'POST', undefined, data);
 }
 
+function patch(endpoint, data) {
+    return call(endpoint, 'PATCH', undefined, data);
+}
+
+//////////////////////
```
Comment (mattmahn) [inline src/lib/api-client.js]: `\n` :sob:

File: src/views/components/edit-model-form.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import ModelController from '../../controllers/components/model';
+import ModelStore  from '../../store/model-store';
+
+export default class EditModelForm extends React.Component {
+
+    constructor(props) {
+        super(props);
+        if (props.model === undefined)
+            this.state = {model: null};
+        else
```
Comment (mattmahn) [inline src/views/components/edit-model-form.jsx]: I dislike the lack of `{ }`.

File: src/views/components/edit-model-form.jsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                manufacturer: props.model.manufacturer,
+                vendor: props.model.vendor,
+                location: props.model.location,
+                price: props.model.price,
+                isFaulty: props.model.isFaulty,
+                faultDescription: props.model.faultDescription
+            };
+    }
+    componentDidMount(){
+        if(this.state.model === null) {
```
Comment (mattmahn) [inline src/views/components/edit-model-form.jsx]: Oh. So you have `{ }` here, but not in the constructor…

File: src/views/components/edit-model-form.jsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        this.setState({
+            price: e.target.value
+        });
+    }
+    changeIsFaulty(e){
+        this.setState({
+            isFaulty: e.target.value
+        });
+    }
+    changeFaultDescription(e) {
```
Comment (mattmahn) [inline src/views/components/edit-model-form.jsx]: I _love_ the consistent whitespace

File: test/unit/lib/api-client.js [hunk 5]
```diff
@@ -315,11 +316,60 @@ describe('API Client', () => {
...
             response
-        })
+        });
         return searchStudent('123456').then(data => {
             assert.deepEqual(data, response.data);
             mockServer.validate();
         });
     });
 
+    it('updateModel', () => {
```
Comment (mattmahn) [inline test/unit/lib/api-client.js]: Nit: "it updateModel" is hardly even a phrase. It should be `describe` (with an `it` inside) not `it`.

**Comment by gearedmonkey** [comment]
@mattmahn I've switched it to query string + json body.

**Pull request merged**